bootstrapper/cmd/bootstrapper/main.go

```diff
@@ -125,6 +125,8 @@ func main() {
 		if err != nil {
 			log.With(zap.Error(err)).Fatalf("Failed to set up cloud logger")
 		}
+		go metadata.PrepareAzureControlPlaneNode(ctx, log)
```

Comment:
I thought about wrapping is in channels to be able to return an error in this function but that also didn't look very nice. Maybe the place is also wrong. Maybe we should add `PrepareControlPlaneModes` as a general function to the CSP interface. This could serve as a hook point for CSP specific hacks.